src/wallet/wallet.cpp

```diff
@@ -585,6 +585,19 @@ void CWallet::ChainTipAdded(const CBlockIndex *pindex,
 {
     IncrementNoteWitnesses(pindex, pblock, sproutTree, saplingTree);
     UpdateSaplingNullifierNoteMapForBlock(pblock);
+
+    // SetBestChain() can be expensive for large wallets, so do this
+    // at most once per hour; the wallet state will be brought up to
+    // date during rescanning on startup.
+    int64_t nNow = GetTimeMicros();
```

Comment:
One objective of mine is to replace all such calls with invocations through an interface such as 
```
#ifndef ZCASH_CLOCK_H
#define ZCASH_CLOCK_H

#include <chrono>

class CClock {
public:
    virtual int64_t GetTime();
}

class SystemClock: public CClock {
public:
    int64_t GetTime() {
        return time(NULL);
    }
}

class FixedClock: public CClock {
private:
    int64_t nFixedTime;

public:
    FixedClock(int64_t nFixedTimeIn) {
        nFixedTime = nFixedTimeIn;
    }

    int64_t GetTime() {
        return nFixedTime;
    }
}

#endif // ZCASH_CLOCK_H
```

and then passing a Clock value explicitly; we can then provide functionality to use an alternate CClock implementation for testing.

Reply:
Also seems like we should be using `clock_gettime(CLOCK_MONOTONIC)` in many places (where we are timing intervals). It's time since boot, higher resolution, and isn't affected by system date and time changes.

But we can address all this in a separate PR or PRs.